engine/src/main/java/com/redhat/cloud/notifications/events/EndpointProcessor.java

```diff
@@ -104,7 +108,17 @@ public void process(Event event, boolean replayEmailsOnly) {
 
             Log.debugf("[org_id: %s] Found %s endpoints for the aggregation event: %s", event.getOrgId(), endpoints.size(), event);
         } else {
-            endpoints = endpointRepository.getTargetEndpoints(event.getOrgId(), event.getEventType());
+            if (engineConfig.isUseDirectEndpointToEventTypeEnabled()) {
+                endpoints = endpointRepository.getTargetEndpointsWithoutUsingBgs(event.getOrgId(), event.getEventType());
+            } else {
+                endpoints = endpointRepository.getTargetEndpoints(event.getOrgId(), event.getEventType());
+                if (engineConfig.isDirectEndpointToEventTypeDryRunEnabled()) {
+                    final List<Endpoint> fetchEndpointWithoutBg = endpointRepository.getTargetEndpointsWithoutUsingBgs(event.getOrgId(), event.getEventType());
+                    if (!endpoints.equals(fetchEndpointWithoutBg)) {
```

Comment:
This could fail if the two lists contain the same endpoints but in a different order. Transforming the lists into sets before running the `equals` would fix that.

Reply:
that's right, thanks